examples/petstore-expanded/chi/petstore_test.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/deepmap/oapi-codegen/examples/petstore-expanded/chi/api"
+)
+
+func DoJson(handler http.Handler, method string, url string, body interface{}) *httptest.ResponseRecorder {
+	rr := httptest.NewRecorder()
+	b, _ := json.Marshal(body)
+	req, _ := http.NewRequest(method, url, bytes.NewBuffer(b))
+	handler.ServeHTTP(rr, req)
+
+	return rr
+}
+
+func TestPetStore(t *testing.T) {
+	var err error
+
+	store := api.NewPetStore()
+	h := api.Handler(store)
+
+	t.Run("Add pet", func(t *testing.T) {
```

Comment:
Hmm, that's an interesting way to structure tests. I've not seen this approach before.